### PR TITLE
Deprecate methods on History [9476]

### PR DESCRIPTION
**Pull request by MiguelCompany**

### CODE REVIEW

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/2761/)
* Linux aarch64[Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux%20aarch64/129/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/2540/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/2773/)

**Review by IkerLuengo — CHANGES_REQUESTED** [review]
More than deprecate, this removes the functionality.
We should mark the methods as deprecated, but keep the functionality until it is finally removed.

File: include/fastdds/rtps/history/History.h [hunk 1]
```diff
@@ -225,6 +235,10 @@ class History
     //!Print the seqNum of the changes in the History (for debuggisi, mng purposes).
```
Comment (IkerLuengo) [inline include/fastdds/rtps/history/History.h:235]: I'm not going to ask for a change due to this, but if another commit is necessary for some reason, can we take the opportunity to correct this documentation here?

**Review by IkerLuengo — APPROVED** [review]
LGTM

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/2801/)
* Linux aarch64[Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux%20aarch64/148/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/2571/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/2813/)

**Pull request merged**